sdk/tables/data-tables/src/TableBatch.ts

```diff
@@ -0,0 +1,294 @@
+   * @param options options for the delete operation
+   */
+  public deleteEntity(
+    partitionKey: string,
+    rowKey: string,
+    options?: DeleteTableEntityOptions
+  ): void {
+    this.checkPartitionKey(partitionKey);
+    this.pendingOperations.push(this.interceptClient.deleteEntity(partitionKey, rowKey, options));
+  }
+
+  /**
+   * Adds an updateEntity operation to the batch
+   * @param entity entity to update
+   * @param mode update mode (Merge or Replace)
+   * @param options options for the update operation
+   */
+  public updateEntity<T extends object>(
+    entity: TableEntity<T>,
+    mode: UpdateMode,
+    options?: UpdateTableEntityOptions
+  ): void {
+    this.checkPartitionKey(entity.partitionKey);
+    this.pendingOperations.push(this.interceptClient.updateEntity(entity, mode, options));
+  }
+
+  /**
+   * Submits the operations in the batch
+   */
+  public async submitBatch(): Promise<any> {
+    await Promise.all(this.pendingOperations);
+    const body = this.batchRequest.getHttpRequestBody();
+    const client = new ServiceClient();
+    const headers: RawHttpHeaders = {
+      accept: "application/json",
+      "x-ms-version": "2019-02-02",
+      "Accept-Charset": "UTF-8",
+      DataServiceVersion: "3.0;",
+      MaxDataServiceVersion: "3.0;NetFx",
+      "Content-Type": `multipart/mixed; boundary=batch_${this.batchGuid}`,
+      Connection: "Keep-Alive"
+    };
+
+    const { span, updatedOptions } = createSpan("TableBatch-submitBatch", {});
+    const request = new WebResource(this.url, "POST", body, undefined, new HttpHeaders(headers));
+
+    request.spanOptions = updatedOptions;
+
+    if (this.credential) {
+      const authHeader = getAuthorizationHeader(request, this.credential);
+      request.headers.set("Authorization", authHeader);
+    }
+
+    const requestOptions: RequestPrepareOptions = {
+      method: "POST",
+      url: this.url,
+      headers: request.headers.rawHeaders(),
+      body,
+      disableJsonStringifyOnBody: true
+    };
+    try {
+      const rawBatchResponse = await client.sendRequest(requestOptions);
+      return parseBatchResponse(rawBatchResponse);
+    } catch (error) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: error.message
+      });
+      throw error;
+    } finally {
+      span.end();
+    }
+  }
+
+  private checkPartitionKey(partitionKey: string): void {
+    if (this.partitionKey !== partitionKey) {
+      throw new Error("All operations in a batch must target the same partitionKey");
+    }
+  }
+}
+
+function parseBatchResponse(response: HttpOperationResponse): TableBatchResponse {
+  const subResponsePrefix = `--changesetresponse_`;
+  const status = response.status;
+  const rawBody = response.bodyAsText || "";
+  const splitBody = rawBody.split(subResponsePrefix); // string after ending is useless
+  const subResponses = splitBody.slice(1, splitBody.length - 1); // string before first response boundary is useless
```

Comment:
To make sure I understand the intention here, we're getting rid of the first and last item of the array, correct? Is that because the last item is an empty string after the final boundary?

Reply:
Yes the first and last are the boundaries

Reply:
I'm adding a comment